core/src/main/scala/org/apache/spark/storage/BlockId.scala

```diff
@@ -124,11 +134,12 @@ class UnrecognizedBlockId(name: String)
 @DeveloperApi
 object BlockId {
   val RDD = "rdd_([0-9]+)_([0-9]+)".r
-  val SHUFFLE = "shuffle_([0-9]+)_([0-9]+)_([0-9]+)".r
+  val SHUFFLE = "shuffle_([0-9]+)_(-?[0-9]+)_([0-9]+)".r
```

Comment:
nit: `\\d+` instead ?

Reply:
This is to support a merged map id which has `-1` as mapId. Is the suggestion to replace `-?` with `\\d+`? That doesn't work or maybe I misunderstood.

Reply:
I was thinking more of `-?\\d+` - but that is ok, let us ignore this given every other regex is using the `[0-9]` anyway.